test/util/DataUtils.spec.ts

```diff
@@ -72,6 +72,15 @@ describe('is functions', () => {
         expect(isNumber(value)).toBe(result);
       });
     });
+
+    it('should allow type refinement', () => {
```

Comment:
This seems like a test of Typescript behaviour itself. Did you intend to commit this? 🤔 If yes, why?

Reply:
I guess its a test that `isNumber` ends up allowing type refinement yeah? But also yeah +1 I don't really see the value - if we know it can do type refinement it never won't be able to

Reply:
So I wrote this test for two reasons:

1. I was struggling with a type error in different file. That error might have been caused by the missing refinement. I wasn't certain if it does behave that way so I wrote a test. And then I didn't delete the test when I was done debugging.
2. I wanted to have an example for the next time someone asks how to write unit tests for types - I remember it came up in the other PR.

That said, I can delete it if we feel like it's taking up too much space.

Reply:
eh I think its fine but maybe we can add a comment that explains it since it is slightly confusing at a glance